dump(v) when `v` is an option value, inside a generic function leads to a cgen error

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 20.04.5 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0

getwd: /v/vnew
vmodules: /home/delian/.vmodules
vroot: /v/vnew
vexe: /v/vnew/v
vexe mtime: 2023-03-07 08:10:13
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.3 1a48d08.784592a

Git version: git version 2.39.0
Git vroot status: weekly.2023.10-2-g784592af-dirty (2 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg a.v`
```v
fn abc[T]() {
	a := []?int{len: 2}
	for v in a {
		dump(v)
	}
}

fn main() {
	abc[int]()
}

```

**What did you expect to see?**

A program that compiles and runs.


**What did you see instead?**
```
==================
/tmp/v_1000/a.12402514639071395678.tmp.c:12315: warning: implicit declaration of function '_v_dump_expr__option_int'
tcc: error: undefined symbol '_v_dump_expr__option_int'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

Notes:
  a) making `abc` non generic, avoids the cgen error.
  b) using `println` and not `dump`, avoids the cgen error.

--- a/vlib/v/gen/c/dumpexpr.v
+++ b/vlib/v/gen/c/dumpexpr.v
@@ -20,7 +20,8 @@ fn (mut g Gen) dump_expr(node ast.DumpExpr) {
 		if node.expr is ast.Ident {
 			// var
 			if node.expr.info is ast.IdentVar && node.expr.language == .v {
-				name = g.typ(g.unwrap_generic(node.expr.info.typ)).replace('*', '')
+				name = g.typ(g.unwrap_generic(node.expr.info.typ.clear_flag(.shared_f).clear_flag(.option).clear_flag(.result))).replace('*',
+					'')
 			}
 		}
 	}

--- a/vlib/v/tests/option_array_dump_in_generic_fn_test.v
+++ b/vlib/v/tests/option_array_dump_in_generic_fn_test.v
@@ -1,0 +1,15 @@
+fn abc[T]() []int {
+	a := []?int{len: 2}
+	mut s := []int{}
+	for v in a {
+		s << dump(v)
+	}
+	return s
+}
+
+fn test_main() {
+	arr := abc[int]()
+	assert arr.len == 2
+	assert arr[0] == 0
+	assert arr[1] == 0
+}

